### ISSUE TITLE
incorrect merging behavior for bind_mounts when specified in template and config file

### ISSUE THREAD

**Issue by zjorgensenbits** [determined-ai/determined#1660]
According to the docs "If the field specifies a list value, the merged value will be the concatenation of the list specified in the template and that specified in the configuration."

However, I've noticed that in the case of bind mounts, the bind_mounts section in the config file completely overwrites the one from the template, rather than concatenating the two lists.

On the other hand, I tested the same thing with the environment_variables key and it works as expected in this case.

**Comment by rb-determined-ai** [comment]
Sounds like a bug.  I'll file a ticket internally.  Thank you for filing it!

You know, just in the last week or two we were were actually talking internally about the templates feature.  How would you rate the templates feature on a scale of 1-10 for both usability and usefulness?  (assuming this bug you are reporting did not exist)

**Comment by zjorgensenbits** [comment]
Thanks! Assuming this bug was fixed, I'd probably give it a 9. I think it's intuitive and works like I'd expect it to work. My team is using the templates feature for several different things: (1) to ease mounting all of our various nfs data stores, which are already mounted in the same place on each of our nodes (2) we have a template for notebooks that mounts one of our nfs shares into the notebook working directory for notebook persistence, (3) we set various environmental variables that are used by some our internal tools, and (4) to generally enforce best practices (e.g., regarding checkpoint storage).

The bind mount merging issue is problematic for us because we have a data download script that expects /tmp_host to map to the host's /tmp directory, which it uses as the download location to prevent multiple trials needing to download the same data on the same host. If someone has a bind mount in their config file, it overwrites the /tmp_host bind mount and causes the download script to just download to the container's /tmp instead (unless the user remembers to explicitly mount it in their config file).

I think the templates feature could by enhanced by having a mechanism to allow one to specify multiple templates for the same job or to be able to reference one template inside another. But those would mostly be for convenience rather than enabling use cases that aren't already covered by the existing templates feature. It could also make things too convoluted, so I'm not sure how much we would make use of such features to be honest.
